### PR TITLE
tools: allow the upstream git remote to have any name

### PR DESCRIPTION
**Pull request by dhellmann**
Instead of requiring that the local clone 'origin' point to the
openshift/enhancements fork, allow any name. Look for the remote with
the right URL when it is needed. Avoid looking for the remote for
every PR by creating a new Summarizer type that holds the remote
information in a way that can be reused for most operations that need
it, other than the initial repository configuration.

### CODE REVIEW

File: tools/enhancements/summary.go [hunk 1]
```diff
@@ -24,29 +24,54 @@ func UpdateGitRepo() error {
 	return nil
 }
 
-const desiredRefSetting = "+refs/pull/*/head:refs/remotes/origin/pr/*"
+// findOpenShiftRemote looks through the remotes for the current git
+// repository to find one with "openshift/enhancements" in the URL
```
Comment (aravindhp) [inline tools/enhancements/summary.go:28]: I suggest adding a requirement to have an upstream remote in the README.

File: tools/report/report.go [hunk 2]
```diff
@@ -148,9 +148,9 @@ func showOnePR(prd *stats.PullRequestDetails, withDescription bool, withLabels b
 
 // ShowPRs prints a section of the report by formatting the
 // PullRequestDetails as a list.
-func ShowPRs(name string, prds []*stats.PullRequestDetails, withDescription bool, withLabels bool) {
+func ShowPRs(summarizer *enhancements.Summarizer, name string, prds []*stats.PullRequestDetails, withDescription bool, withLabels bool) error {
```
Comment (aravindhp) [inline tools/report/report.go:151]: Is there a reason to return `error`? I don't see an actual error being returned nor is it checked for in the call site.

**Review by aravindhp — COMMENTED** [review]
Thanks for fixing this, @dhellmann 

Tested-by: Aravindh Puthiyaparambil <aravindh@redhat.com>

File: tools/enhancements/summary.go (hunk 1 above)
Comment (dhellmann) [reply tools/enhancements/summary.go:28]: Good idea. I added another commit to this PR to do that.

File: tools/report/report.go (hunk 2 above)
Comment (dhellmann) [reply tools/report/report.go]: Ah, one iteration had this function creating the summarizer and then I moved that and didn't clean up. Fixed.

**Pull request merged**